source/developers/development_environment.markdown

```diff
@@ -11,20 +11,55 @@ footer: true
 
 You'll need to set up a development environment if you want to develop a new feature or component for Home Assistant. Read on to learn how to set up.
 
-* Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
-
-* Consider setting up a virtual environment using [`venv`](https://docs.python.org/3.4/library/venv.html) before running the setup script.
+### {% linkable_title Setup Local Repository %}
 
+Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
+Once forked, setup your local copy of the source using the commands:
 ```bash
 $ git clone https://github.com/YOUR_GIT_USERNAME/home-assistant.git
 $ cd home-assistant
 $ git remote add upstream https://github.com/home-assistant/home-assistant.git
+```
+
+### {% linkable_title Prepare Your Environment %}
+
+#### {% linkable_title Core dependencies %} 
+
+Install the core dependencies.
+```bash
+$ sudo apt-get install python3-pip python3-dev libssl-dev
+```
+
+#### {% linkable_title Frontend dependencies (optional) %} 
```

Comment:
@kellerza If script/setup is needed/expected to run run whenever you do a pull, then it we could add a section on updating here, but perhaps that would better belong on the page: https://home-assistant.io/developers/development_catching_up/

You can run and re-run the setup script, I did ;-) Whilst I was working through different options for pulling Node into the venv. As part of another PR that @armills worked on.

The goal of this PR is to ideally set everything up for a new developer, if we can avoid them/me seeing errors when following the published instructions. A developer who knows enough and doesn't want to install node may be happy to see an error during setup and be able to understand the impact.